docs/concept-map.md

```diff
@@ -0,0 +1,122 @@
+   which exposes admin control over the active schema of `DRIVER` records
+
+2. A dashboard web app, implemented as an AngularJS app in `DRIVER/web`, which
+   allows users to browse, filter, and download geospatial data 
+
+3. A standalone Django app, `ashlar`, which exposes a RESTful CRUD API
+   for retrieving and updating both A) geospatial records and B) the schema
+   composing those geospatial records
+
+4. A standalone Django app, `djsonb`, which provides a Python API for
+   performing queries against arbitrarily nested Postgres JSONB fields in
+   Django
+
+5. A data collection mobile app, implemented as an Android app in `DRIVER`,
+   which produces forms for saving records based on the active schema of
+   `DRIVER` records
+
+In my understanding, components 1, 2, and 5 (the schema editor, dashboard, and
+mobile app) interoperate through component 3 (`ashlar`) to update and retrieve
+records and manage schema for those records. Component 4 (`djsonb`) provides the Python API
+for filtering and querying those records, while `ashlar` provides a RESTful
+version of this API. 
+
+## Future directions
+
+Based on the 5 components above, I can see a few possible futures for Ashlar,
+all of which seem compelling but each of which involves moving the project in
+a different direction:
+
+1. Ashlar as an **integrated development toolkit for building apps with
+   flexible schemas.** This future seems to be the closest thing to the
+   direction that development is currently moving on Ashlar/DRIVER. In this
+   future, Ashlar is likely still a Django app, although possibly with more
+   backends than just Postgres; it provides an admin backend as well as
+   automatic forms for managing schemas. It would be tightly integrated with its
+   implementation, opinionated, and a relatively heavy lift to install/deploy,
+   although at the same time it would offer the most features in one package.
+   It may have an Android toolkit sitting on top of it, too.
+
+1. Ashlar as a **lightweight, JSON-backed database server with dynamic schemas**.
+   This future is further away from the present, and involves a slightly more
+   abstracted version of Ashlar. In this future, Ashlar wouln't have have to be a 
+   Django app; it would  simply be a database server that speaks JSON, both on the
+   level of creating/updating records and also managing schema. It could be
+   implemented in anything that can talk like a database server.
+
+2. Ashlar as a **data model specification, with many possible implementations**.
+   This future is the furthest away, and also the most abstract. Ashlar would be
+   a specification for managing flexible schemas and records, and nothing more. 
+   It could have implementations in a variety of different languages and frameworks.
+
+In addition to these directions for Ashlar, I also see a few possibilities for
+separate repos:
+
+4. Some portion of djsonb as a **simple syntax for querying JSON over a REST API**. 
```

Comment:
GraphQL seems to be gaining more traction these days; I don't know a lot about it myself but a few folks here have looked into it and seem enthusiastic. I wonder if Ashlar + GraphQL could be an interesting combination.